packages/@tailwindcss-cli/src/commands/build/index.ts

```diff
@@ -218,15 +215,28 @@ export async function handle(args: Result<ReturnType<typeof options>>) {
             args['--input'] ?? base,
           )
 
-          build = compile(input).build
-          compiledCss = build(candidates)
+          // Create a new compiler, given the new `input`
+          compiler = compile(input)
+
+          // Re-scan the directory to get the new `candidates`
+          scanDirResult = scanDir({ base, contentPaths: compiler.globs, outputGlobs: true })
+
+          // Setup new watchers
+          cleanupWatchers = await createWatchers(scanDirResult.globs, handle)
+
+          // Re-compile the CSS
+          compiledCss = compiler.build(scanDirResult.candidates)
         }
 
         // Scan changed files only for incremental rebuilds.
         else if (rebuildStrategy === 'incremental') {
-          let newCandidates = scanFiles(changedFiles, IO.Sequential | Parsing.Sequential)
+          let newCandidates = scanDirResult.scanFiles(changedFiles)
```

Comment:
Attached this `scanFiles` method to the `scanDirResult` such that we don't have to pass in all the globs to ensure that a file is allowed or not.

This way we can keep the state (the globs) in Rust

Reply:
Beautiful!